pkg/openstack/loadbalancer.go

```diff
@@ -2910,3 +3176,42 @@ func GetLoadBalancerSourceRanges(service *corev1.Service) (netsets.IPNet, error)
 	}
 	return ipnets, nil
 }
+
+// PreserveGopherError preserves the error details delivered with the response
+// that are explicitly omitted by dedicated error types.
+// error types from provider_client.go
+func PreserveGopherError(rawError error) error {
```

Comment:
What's the problem does this function solve? Any examples?

Reply:
I added a more detailed comment and an example

Reply:
@mandelsoft  I suppose this func could be moved to https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/util/errors/errors.go